src/components/AllocationsComponent.jsx

```diff
@@ -14,50 +14,30 @@ import '../_css/AllocationsComponent.css';
 
 export class AllocationsComponent extends Component {
   state = {
-    currentAllocations: [],
-    offset: 0,
     activePage: 1,
     limit: 10
   }
 
   componentDidMount() {
-    this.props.loadAllocationsAction();
-    this.setTableContent();
+    this.props.loadAllocationsAction(this.state.activePage);
   }
 
-  componentDidUpdate(prevProps) {
-    if (this.props.allAllocations.length !== prevProps.allAllocations.length) {
-      this.setTableContent();
-    }
-  }
-
-  getTotalPages = () => Math.ceil(this.props.allAllocations.length / this.state.limit);
+  getTotalPages = () => Math.ceil(this.props.allocationsCount / this.state.limit);
 
-  setTableContent = () => {
-    const currentAllocations = this.props.allAllocations.slice(
-      this.state.offset,
-      (this.state.activePage * this.state.limit)
-    );
-    // format date
+  formatDate = (timeStamp) => {
     const dateOptions = {
       weekday: 'long',
       year: 'numeric',
       month: 'long',
       day: 'numeric'
     };
-    currentAllocations.map(el => el.formatted_date = new Date(el.created_at).toLocaleDateString('en-US', dateOptions));
-    this.setState({
-      currentAllocations
-    });
+    return new Date(timeStamp).toLocaleDateString('en-US', dateOptions);
   }
 
-  calculateOffset = (activePage, limit) => (activePage - 1) * limit;
-
   handlePaginationChange = (event, { activePage }) => {
     this.setState({
-      activePage,
-      offset: this.calculateOffset(activePage, this.state.limit)
-    }, () => this.setTableContent());
+      activePage
+    }, () => this.props.loadAllocationsAction(this.state.activePage));
```

Comment:
`loadAllocationsAction` is asynchronous, therefore there's no need to use the setState callback. `handlePaginationChange`can be re-written to:
```
handlePaginationChange = (event, { activePage }) => {
  this.setState({ activePage });
  this.props.loadAllocationsAction(activePage);
}
```